File: The Descent/The Descent.cpp
//Answer to The Descent in C++

#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int main()
{
    int k=0,l=0;
    while (1) {
        for (int i = 0; i < 8; i++) {
            int mountainH; // represents the height of one mountain.
            cin >> mountainH; cin.ignore();
            if(mountainH>k)
            {
                k=mountainH;
                l=i;
            }
        }

        cout << l << endl; // The index of the mountain to fire on.
        k=0;
        l=0;
    }
}

File: The Descent/The Descent.py
#Answer to The Descent in Python

import sys
import math

while True:
    j = 0
    l = 0
    for i in range(8):
        mountain_h = int(input())  # represents the height of one mountain.
        if mountain_h > j:
            j=mountain_h
            l=i
    print(l)
    j=0
    l=0
